src/Features/LanguageServer/Protocol/Handler/Diagnostics/WorkspacePullDiagnosticHandler.cs

```diff
@@ -56,71 +58,124 @@ protected override DiagnosticTag[] ConvertTags(DiagnosticData diagnosticData)
             return ConvertTags(diagnosticData, potentialDuplicate: true);
         }
 
-        protected override ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(RequestContext context, CancellationToken cancellationToken)
-            => GetDiagnosticSourcesAsync(context, GlobalOptions, cancellationToken);
+        protected override async ValueTask<ImmutableArray<IDiagnosticSource>> GetOrderedDiagnosticSourcesAsync(
+            VSInternalWorkspaceDiagnosticsParams diagnosticsParams,
+            RequestContext context,
+            CancellationToken cancellationToken)
+        {
+            // If we're being called from razor, we do not support WorkspaceDiagnostics at all.  For razor, workspace
+            // diagnostics will be handled by razor itself, which will operate by calling into Roslyn and asking for
+            // document-diagnostics instead.
+            if (context.ServerKind == WellKnownLspServerKinds.RazorLspServer)
+                return ImmutableArray<IDiagnosticSource>.Empty;
```

Comment:
@dibarbet we both check this *and* below check `if (document.IsRazorDocument())` to bail out.  Do we need both?  Or can we skip the second check if the server is the razor server?

Reply:
I believe we need both - there are two cases
1.  The razor C# server is being asked for workspace diagnostics (this check here).  The razor C# server shouldn't give anything back
2.  The normal C# server is being asked for workspace diagnostics, and it sees a Razor document.  It should not return diagnostics for that document because we rely on razor to do a document pull for all razor documents.